serde/src/main/java/io/datacater/core/serde/JsonDeserializer.java

```diff
@@ -4,14 +4,16 @@
 import io.vertx.core.buffer.Buffer;
 import io.vertx.core.json.JsonObject;
 import org.apache.kafka.common.serialization.Deserializer;
+import java.util.Map;
 
-public class JsonDeserializer implements Deserializer<JsonObject> {
+public class JsonDeserializer implements Deserializer<Map<String, Object>> {
```

Comment:
Shall we rename this to `MapDeserializer`? There might be a point to actually use `JsonObject` internally.

Reply:
If we would deserialize to a JsonObject, we would end up with the following string representation:

```
{
  "is_empty": true|false,
  "map": {}
```

We are calling [getMap()](https://vertx.io/docs/apidocs/io/vertx/core/json/JsonObject.html#getMap--) on the `JsonObject` to get the map directly, without the information on emptiness.

I believe this is almost never useful for us. Do you have an use case in mind where we would prefer to work with `JsonObject`?